tests/integrated/test-drift-instability/2fluid.cxx

```diff
@@ -3,415 +3,420 @@
+
+  // 3D evolving fields
+  Field3D rho, Te, Ni, Ajpar, Vi, Ti;
+
+  // Derived 3D variables
+  Field3D phi, Apar, Ve, jpar;
+
+  // Non-linear coefficients
+  Field3D nu, mu_i, kapa_Te, kapa_Ti;
+
+  // 3D total values
+  Field3D Nit, Tit, Tet, Vit;
+
+  // pressures
+  Field3D pei, pe;
+  Field2D pei0, pe0;
+
+  // Metric coefficients
+  Field2D Rxy, Bpxy, Btxy, hthe;
+
+  // parameters
+  BoutReal Te_x, Ti_x, Ni_x, Vi_x, bmag, rho_s, fmei, AA, ZZ;
+  BoutReal lambda_ei, lambda_ii;
+  BoutReal nu_hat, mui_hat, wci, nueix, nuiix;
+  BoutReal beta_p;
+
+  // settings
+  bool estatic, ZeroElMass; // Switch for electrostatic operation (true = no Apar)
+  BoutReal zeff, nu_perp;
+  bool evolve_rho, evolve_te, evolve_ni, evolve_ajpar, evolve_vi, evolve_ti;
+  BoutReal ShearFactor;
+
+  // Inversion objects
+  std::unique_ptr<Laplacian> phi_solver;
+  std::unique_ptr<Laplacian> apar_solver;
+
+  FieldGroup comms; // Group of variables for communications
+
+  Coordinates* coord; // Coordinate system
+
+  CELL_LOC maybe_ylow;
+
+protected:
+  int init(bool UNUSED(restarting)) override {
+    Field2D I; // Shear factor
+
+    output.write("Solving 6-variable 2-fluid equations\n");
+
+    /************* LOAD DATA FROM GRID FILE ****************/
+
+    // Load 2D profiles (set to zero if not found)
+    GRID_LOAD(Ni0);
+    GRID_LOAD(Ti0);
+    GRID_LOAD(Te0);
+    GRID_LOAD(Vi0);
+    GRID_LOAD(Ve0);
+    GRID_LOAD(phi0);
+    GRID_LOAD(rho0);
+    GRID_LOAD(Ajpar0);
+
+    // Load magnetic curvature term
+    b0xcv.covariant = false;  // Read contravariant components
+    mesh->get(b0xcv, "bxcv"); // b0xkappa terms
+
+    // Coordinate system
+    coord = mesh->getCoordinates();
+
+    // Load metrics
+    GRID_LOAD(Rxy);
+    GRID_LOAD(Bpxy);
+    GRID_LOAD(Btxy);
+    GRID_LOAD(hthe);
+    mesh->get(coord->dx, "dpsi");
+    mesh->get(I, "sinty");
+
+    // Load normalisation values
+    GRID_LOAD(Te_x);
+    GRID_LOAD(Ti_x);
+    GRID_LOAD(Ni_x);
+    GRID_LOAD(bmag);
+
+    Ni_x *= 1.0e14;
+    bmag *= 1.0e4;
+
+    /*************** READ OPTIONS *************************/
+
+    // Read some parameters
+    Options* globalOptions = Options::getRoot();
+    Options* options = globalOptions->getSection("2fluid");
+    OPTION(options, AA, 2.0);
+    OPTION(options, ZZ, 1.0);
+
+    OPTION(options, estatic, false);
+    OPTION(options, ZeroElMass, false);
+    OPTION(options, zeff, 1.0);
+    OPTION(options, nu_perp, 0.0);
+    OPTION(options, ShearFactor, 1.0);
+
+    (globalOptions->getSection("Ni"))->get("evolve", evolve_ni, true);
+    (globalOptions->getSection("rho"))->get("evolve", evolve_rho, true);
+    (globalOptions->getSection("vi"))->get("evolve", evolve_vi, true);
+    (globalOptions->getSection("te"))->get("evolve", evolve_te, true);
+    (globalOptions->getSection("ti"))->get("evolve", evolve_ti, true);
+    (globalOptions->getSection("Ajpar"))->get("evolve", evolve_ajpar, true);
+
+    if (ZeroElMass)
```

Comment:
warning: statement should be inside braces [readability-braces-around-statements]

```cpp
if (ZeroElMass)
               ^
                {
```